src/game-engine/world/action/magic-on-npc.action.ts

```diff
@@ -0,0 +1,76 @@
+import { Player } from '@engine/world/actor/player/player';
+import { ActionHook, getActionHooks } from '@engine/world/action/hooks';
+import { advancedNumberHookFilter, questHookFilter } from '@engine/world/action/hooks/hook-filters';
+import { ActionPipe, RunnableHooks } from '@engine/world/action/index';
+import { Npc } from '../actor/npc/npc';
+
+
+/**
+ * Defines a button action hook.
+ */
+export interface Magic_On_NPCActionHook extends ActionHook<Magic_On_NPCAction, magic_on_npcActionHandler> {

```

Comment:
Only request atm is to use `MagicOnNpcActionHook` naming as opposed to using underscores in class/interface names

Reply:
Refactored underscores out in latest commit #13f1108